app/services/ChargeHistoryService.scala

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services
+
+import auth.MtdItUser
+import connectors.ChargeHistoryConnector
+import models.chargeHistory._
+import models.financialDetails.DocumentDetail
+import uk.gov.hmrc.http.HeaderCarrier
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class ChargeHistoryService @Inject()(chargeHistoryConnector: ChargeHistoryConnector) {
+
+  def chargeHistoryResponse(isLatePaymentCharge: Boolean, isPayeSelfAssessment: Boolean, chargeReference: Option[String],
+                                    isChargeHistoryEnabled: Boolean, isCodingOutEnabled: Boolean)
+                                   (implicit user: MtdItUser[_], hc: HeaderCarrier, ec: ExecutionContext): Future[Either[ChargeHistoryResponseModel, List[ChargeHistoryModel]]] = {
+    if (!isLatePaymentCharge && isChargeHistoryEnabled && !(isCodingOutEnabled && isPayeSelfAssessment)) {
```

Comment:
I would suspect you were intended to defined this method in this way ?
with the next signature, to return first error if any or list of values in the List 
 
_Future[Either[ChargesHistoryErrorModel, List[ChargeHistoryModel]]]_

```
 def chargeHistoryResponse(isLatePaymentCharge: Boolean, isPayeSelfAssessment: Boolean, chargeReference: Option[String],
                                    isChargeHistoryEnabled: Boolean, isCodingOutEnabled: Boolean)
                                   (implicit user: MtdItUser[_], hc: HeaderCarrier, ec: ExecutionContext): Future[Either[ChargesHistoryErrorModel, List[ChargeHistoryModel]]] = {
    if (!isLatePaymentCharge && isChargeHistoryEnabled && !(isCodingOutEnabled && isPayeSelfAssessment)) {
      chargeHistoryConnector.getChargeHistory(user.nino, chargeReference).map {
        case chargesHistory: ChargesHistoryModel => Right(chargesHistory.chargeHistoryDetails.getOrElse(Nil))
        case errorResponse : ChargesHistoryErrorModel => Left(errorResponse)
      }
    } else {
      Future.successful(Right(Nil))
    }
  }
```

Reply:
+ conditions in the if statement is hard to read / and understand, we can refactor this later, but any condition were we have more than 2 values in the predicate we need to simplify in possible